loginModuleJava/src/main/java/com/brandon/loginmodulejava/view/LoginActivity.java

```diff
@@ -0,0 +1,128 @@
+package com.brandon.loginmodulejava.view;
+
+import static com.brandon.loginmodulejava.viewmodel.LoginUiStates.LOADING;
+import static com.brandon.loginmodulejava.viewmodel.LoginUiStates.LOGGED_IN;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.brandon.corejava.utilities.CommonActivityFunctions;
+import com.brandon.corejava.utilities.NotificationDialog;
+import com.brandon.loginmodulejava.databinding.LoginScreenBinding;
+import com.brandon.loginmodulejava.viewmodel.LegacyLoginViewModel;
+import com.brandon.loginmodulejava.viewmodel.LoginUiStates;
+import com.google.android.material.textfield.TextInputEditText;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import dagger.hilt.android.AndroidEntryPoint;
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+@AndroidEntryPoint
+public class LoginActivity extends AppCompatActivity implements CommonActivityFunctions {
+    @Inject
+    LegacyLoginViewModel viewModel;
+    private Disposable disposable;
+    private LoginScreenBinding binding;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = LoginScreenBinding.inflate(getLayoutInflater());
+        viewModel = new ViewModelProvider(this).get(LegacyLoginViewModel.class);
+        setContentView(binding.getRoot());
+
+        final TextInputEditText emailField = binding.loginEmailField;
+        final TextInputEditText codeField = binding.loginCodeField;
+
+        onTextChanged(emailField, (String s) -> viewModel.emailChanged(s));
+
+        binding.loginButton.setOnClickListener(v -> {
+            final TextView fieldsError = binding.loginRequiredFieldsError;
+            fieldsError.setVisibility(View.GONE);
+            String email = Objects.requireNonNull(emailField.getText()).toString();
+            String code = Objects.requireNonNull(codeField.getText()).toString();
+            if (viewModel.checkInternetConnection()) {
+                if (!email.isEmpty()) {
+                    if (!code.isEmpty()) {
+                        viewModel.loginClicked(email, code);
+                    } else {
+                        viewModel.loginClicked(email, null);
+                    }
+                } else {
+                    fieldsError.setVisibility(View.VISIBLE);
+                }
+            } else {
+                new NotificationDialog().createNotification(this, getString(com.brandon.uicore.R.string.core_error_no_internet_connection), getString(com.brandon.uicore.R.string.core_error_connect_and_try_again), null);
```

Comment:
Import R?
Also format this better it's one giant line

Reply:
Weird it imported that way, also formatted these betters and properly imported it.